lib/parser/unparse.cc

```diff
@@ -2060,29 +2060,25 @@ class UnparseVisitor {
     }
   }
   void Unparse(const OmpBlockDirective &x) {
-    std::visit(
-        common::visitors{
-            [&](const OmpBlockDirective::Master &y) { Word("MASTER"); },
-            [&](const OmpBlockDirective::Ordered &) { Word("ORDERED "); },
-            [&](const OmpBlockDirective::ParallelWorkshare &) {
-              Word("PARALLEL WORKSHARE ");
-            },
-            [&](const OmpBlockDirective::Parallel &) { Word("PARALLEL "); },
-            [&](const OmpBlockDirective::TargetData &) {
-              Word("TARGET DATA ");
-            },
-            [&](const OmpBlockDirective::TargetParallel &) {
-              Word("TARGET PARALLEL ");
-            },
-            [&](const OmpBlockDirective::TargetTeams &) {
-              Word("TARGET TEAMS ");
-            },
-            [&](const OmpBlockDirective::Target &) { Word("TARGET "); },
-            [&](const OmpBlockDirective::Taskgroup &) { Word("TASKGROUP "); },
-            [&](const OmpBlockDirective::Task &) { Word("TASK "); },
-            [&](const OmpBlockDirective::Teams &) { Word("TEAMS "); },
-        },
-        x.u);
+    switch (x.v) {
```

Comment:
What's the advantage of converting to a enum?

Reply:
Enum classes are simple, safe, get the job done, and you can use `switch` statements on them.  Using `std::variant` on a pile of empty class types violates the principle of least astonishment, forces the use `std::visit` / `std::holds_alternative` to decode, and adds no value.